File: 加密保护/WinLicense v2.1.0.10/WinlicenseSDK/ExamplesSDK/Generators(License,Pass,TrialExt)/PasswordGenerator/C/CBuilder/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include <stdio.h>
#include "..\..\..\..\..\Include\C\WinlicenseSDK.h"
#pragma comment (lib,"..\\..\\..\\..\\..\\Lib\\OMF\\WinlicenseSDK.lib")


//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
    PasswordHash = "kd7dkcdsflkjdhd63urbcmchGFFJ6sJFDBXKLAIO673jdehf7HFTGdfndj73jfkalasdj8thfgh73k394746sdfjhds6364odfjkhdsf86346kjfhf3";
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{
    char    PasswordBuffer[100];
    char*   NameString = new char[NameEdit->Text.Length()+1];
    char    PassBuffer[100];

    strcpy(NameString, NameEdit->Text.c_str() );

    // create a password for the current user

    WLGenPassword(PasswordHash, NameString, PasswordBuffer);

    strcpy(PassBuffer, PasswordBuffer);
    
    PassEdit->Text = PassBuffer;

    delete NameString;
  
}
//---------------------------------------------------------------------------

File: 加密保护/WinLicense v2.1.0.10/WinlicenseSDK/ExamplesSDK/Generators(License,Pass,TrialExt)/PasswordGenerator/Delphi/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, WinlicenseSDK, StdCtrls;

const
PasswordHash = '432ab78bd7ar324ghjd87feg63vksyb375am35fj50abloqnxc543jd73hrgf5289d';

type
    TForm1 = class(TForm)
    Button1: TButton;
    Label1: TLabel;
    Label2: TLabel;
    NameEdit: TEdit;
    PassEdit: TEdit;
    procedure Button1Click(Sender: TObject);

  private
    { Private declarations }

  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);

var

    PasswordBuffer: ARRAY[0..100] of char;

begin

    WLGenPassword(PasswordHash, PChar(NameEdit.Text), PasswordBuffer);
    PassEdit.Text := PasswordBuffer;

end;

end.
